### PR TITLE
EQL: Deprecate lenient sequence declaration

### LINKED ISSUE #55023: EQL: Deprecate alternate sequence params declaration
For simplicity, it was agreed to not be lenient and allow only one way to express.
However for compatibility reasons the grammar would not be changed but the parser would allow only one variant and offer guidance for the second.

Relates to #54597

### PR DESCRIPTION
**Pull request by costin**
Deprecate alternative sequence parameter declaration (with then by)
Disallow maxspan time units missing.

Fix #55023
Relate #54680

### CODE REVIEW

File: x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/parser/LogicalPlanBuilder.java [hunk 1]
```diff
@@ -173,45 +178,40 @@ public TimeValue visitSequenceParams(SequenceParamsContext ctx) {
...
+                    timeUnit = TimeUnit.MINUTES;
+                    break;
+                case "h":
+                    timeUnit = TimeUnit.HOURS;
+                    break;
+                case "d":
+                    timeUnit = TimeUnit.DAYS;
+                    break;
+                default:
+                    throw new ParsingException(source(ctx.timeUnit().IDENTIFIER()),
```
Comment (rw-access) [inline x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/parser/LogicalPlanBuilder.java:205]: I think for #54760, we'll need to add `micros` and `nanos`.

**Review by rw-access — APPROVED** [review]
only q was if we should add additional units for `micros` and `nanos` now, or later.
otherwise LGTM

**Comment by costin** [comment]
@rw-access It's easy to add them but since this is for maxspan, are there sequences that are micro or nanos dependent and where a millisecond would not work?

**Comment by rw-access** [comment]
Milliseconds is sufficient for all of our current rules and all use cases that I'm aware of. Reliable sub-millisecond granularity is often difficult across data sources.

I was mostly wondering for the sake of completion or for other domains.

**Pull request merged**